pallets/proofs-dealer/src/utils.rs

```diff
@@ -689,10 +689,63 @@ impl<T: pallet::Config> ProofsDealerInterface for Pallet<T> {
         )
     }
 
-    // TODO: Add `initialise_provider` method to be called by the FileSystem pallet.
-    // TODO: when a file is first uploaded to a BSP or bucket.
-    // TODO: It would set `LastTickProviderSubmittedProofFor` to the current tick and
-    // TODO: the deadline for submitting a proof in `ChallengeTickToChallengedProviders`.
+    fn initialise_provider(who: &Self::ProviderId) -> DispatchResult {
```

Comment:
Maybe we should rename this to `start_providing`. I wouldn't have guessed that the ticks for challenges was affected here in any way.

Reply:
How about `initialise_provider_challenge_cycle` or `initialise_challenge_cycle`?